### PR TITLE
fix: Templates UI polishing

### LINKED ISSUE #11634: [Task] Polishing templates epic for phase 1.
Deploy preview: https://appsmith-79edqojiy-get-appsmith.vercel.app/

- [x] Update template and application import fork icons
- [x] Change the fork app modal to a similar design as the fork templates modal.
- [x] The cards on the first row should be of the same height(the height of the bigger one)
- [x] The tags in the smaller cards should be of 14px instead of 12px and for subtext it should be 13px instead of 12px
- [x] The image section height in the larger cards should be fixed regardless of the image height.
- [x] The padding between back button and title is reduced to 12px
- [x] The font-size of subtext in all the sections is reduced to 16px ( Function , Industry, Datasources )
- [x] "Showing all 5 templates" font size altered to 20px 
- [x] Show filters which have templates associated with it.
- [x] Show filter count alongside each category

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

Fixes #11634

Changes include 
- Spacing, font size changes in templates
- The fork app modal ui is changed to how the import template modal looks like
- We only show those filters which have templates associated with it.

The many commits are because I created this MR from https://github.com/appsmithorg/appsmith/pull/10964 before it was merged.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manual. Cypress tests for template imports exist already.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/template-height 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.91 **(0)** | 37.26 **(-0.01)** | 35.29 **(-0.01)** | 56.24 **(0)**
 :red_circle: | app/client/src/ce/constants/messages.ts | 76.9 **(-0.01)** | 100 **(0)** | 31.47 **(-0.04)** | 82.1 **(-0.08)**
 :red_circle: | app/client/src/components/ads/DialogComponent.tsx | 86.84 **(0)** | 60 **(-3.33)** | 84.21 **(0)** | 86.11 **(0)**
 :green_circle: | app/client/src/components/ads/Icon.tsx | 99.41 **(0.01)** | 84.21 **(0)** | 100 **(0)** | 99.4 **(0)**
 :red_circle: | app/client/src/pages/Applications/ForkApplicationModal.tsx | 83.33 **(-0.88)** | 31.82 **(-14.33)** | 37.5 **(-5.36)** | 82.5 **(-0.83)**
 :red_circle: | app/client/src/pages/Applications/ForkModalStyles.ts | 87.5 **(-7.95)** | 87.5 **(0)** | 25 **(-25)** | 87.5 **(-7.95)**
 :red_circle: | app/client/src/reducers/uiReducers/templateReducer.ts | 35.71 **(-2.75)** | 100 **(0)** | 0 **(0)** | 35.71 **(-2.75)**</details>

### CODE REVIEW

File: app/client/src/pages/Applications/ForkModalStyles.ts [hunk 1]
```diff
@@ -41,6 +38,8 @@ const OrganizationList = styled.div`
 
 const ButtonWrapper = styled.div`
   display: flex;
+  margin-top: ${(props) => props.theme.spaces[11]}px;
+  gap: ${(props) => props.theme.spaces[4]}px;
```
Comment (arunvjn) [inline app/client/src/pages/Applications/ForkModalStyles.ts:42]: Can you double check if "gap" for all browsers ?

File: app/client/src/pages/Templates/Filters.tsx [hunk 2]
```diff
@@ -203,24 +180,25 @@ function FilterCategory({
...
-          className={`more ${selectedFilters.length && expand && "hide"}`}
-          onClick={toggleExpand}
-          type={TextType.BUTTON_SMALL}
-          underline
-        >
-          {expand
-            ? `- ${createMessage(SHOW_LESS)}`
-            : `+ ${filterList.slice(3).length} ${createMessage(MORE)}`}
-        </Text>
+        {!!filterList.slice(3).length && (
```
Comment (arunvjn) [inline app/client/src/pages/Templates/Filters.tsx:183]: Can you add a comment here on the usage of 3?

File: app/client/src/pages/Applications/ForkModalStyles.ts (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/pages/Applications/ForkModalStyles.ts:42]: Yup had some doubt on safari. Checked the same it works.  The rest of the major ones work too. Ignoring IE here, it looks like it doesn't work there.
https://caniuse.com/flexbox-gap

File: app/client/src/pages/Templates/Filters.tsx (hunk 2 above)
Comment (akash-codemonk) [reply app/client/src/pages/Templates/Filters.tsx]: Added a comment it indicates the number of filters which are shown initially the rest are hidden behind a show more option.

**Review by arunvjn — APPROVED** [review]

**Pull request merged**